src/CsvFile.php

```diff
@@ -340,7 +340,9 @@ public function rowToStr(array $row)
     {
         $return = [];
         foreach ($row as $column) {
-            if (!is_scalar($column) && !is_null($column)) {
+            if (!(is_scalar($column) || is_null($column) ||
```

Comment:
Byl bych pro to vyextrahovat do popsaných proměnných `$isScalar`, `$isNull`, `$isObjectConvertibleToString`, výsledek do `$isStringable` a ten pak jen `!$isStringable`, takhle je ten boolean výraz docela nepřehledný. 

Kdybys byl vyloženě proti, tak aspoň zalámat každou úroveň zvlášť.

Reply:
Nebo to vytáhnout do metody? 
```php
private function isSupportedTypeForWrite($value)
	{
		if (is_scalar($value)) {
			return true;
		}

		if (is_null($value)) {
			return true;
		}

		if (is_object($value)  && method_exists($value, '__toString')) {
			return true;
		}

		return false;
	}
```

Reply:
Tak prosím 

```
if (!(
    is_scalar($column)
    || is_null($column)
    || (
        is_object($column) 
        && method_exists($column, '__toString')
    )
)) {
```

nebo aspoň

```
if (!(
    is_scalar($column)
    || is_null($column)
    || (is_object($column) && method_exists($column, '__toString'))
)) {
```

Reply:
Teoreticky by se dal udělat early exit a sólo podmínky, ale to by čitelnosti moc nepřispělo asi.